### PR TITLE
Fix bug 391: Archive symlinks to folders

### PR DESCRIPTION
**Pull request by davigar15**
Basically I'm checking if the dirnames are links or not. If they are, then I write them to the zip.

### CODE REVIEW

**Comment by SimonRichardson** [comment]
@davigar15 can I get some tests for this, so we don't regress in the future.

File: tests/integration/test_model.py [hunk 1]
```diff
@@ -93,6 +93,25 @@
             assert app in model.applications
 
 
+@base.bootstrapped
+@pytest.mark.asyncio
+async def test_deploy_local_charm_operator_framework(event_loop):
+    from pathlib import Path
```
Comment (SimonRichardson) [inline tests/integration/test_model.py]: If you remove this to the top of the file, we can drop the E402 changes to flake8?

**Comment by davigar15** [comment]
That was meant for the arc/charm.py in the charm example.

We need to append the lib in order to import the objects in ops.

Since we do sys.append(‘lib’) before  actually importing the stuff in ops,
there’s no way we can avoid that addition in flake8.

David

**Comment by SimonRichardson** [comment]
@davigar15 having more thought on this is, can we replicate this without the operator framework. Unfortunately, I think adding the operator framework is a blocker on this.

Additionally, we've got some 2.8 code in master that we don't want to release until later on in the cycle, so a release of pylibjuju isn't going to happen until then.

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by davigar15** [comment]
You’re welcome 😇

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**